From 8420c6efa636598e5451f0b251d302368e359629 Mon Sep 17 00:00:00 2001
From: Holger Freyther <ich@holger-freythers-macbook-pro.local>
Date: Mon, 19 Jan 2009 01:50:09 +0100
Subject: [PATCH 16/22] Add onload javascript foo to mangle the content...

- Remove links to Image: articles (assuming my path layout in the URI)
- Remove links to show/hide buttons... we are not interactive
- Remove relative space divs like in the Taipei article they lead
  to overlapping text on articles...
---
 skins/MonoBook.php |   24 ++++++++++++++++++++++++
 1 files changed, 24 insertions(+), 0 deletions(-)

diff --git a/skins/MonoBook.php b/skins/MonoBook.php
index 3dc0f0b..78e80c8 100644
--- a/skins/MonoBook.php
+++ b/skins/MonoBook.php
@@ -122,6 +122,30 @@ class MonoBookTemplate extends QuickTemplate {
 
 -->
 <?php endif; ?>
+<script type="text/javascript">
+// Hide all images
+var nodes = document.querySelectorAll("a")
+for (i = 0; i < nodes.length; ++i) {
+    if (nodes[i].href.indexOf("/mediawiki/index.php/Image:") != -1) {
+        nodes[i].style.setProperty("display", "none");
+    }
+}
+
+// Hidle all toggle actions show/hide
+var nodes = document.querySelectorAll("a.NavToggle")
+for (i = 0; i < nodes.length; ++i) {
+    nodes[i].style.setProperty("display", "none");
+}
+
+// Remove relative spaced content...
+var nodes = document.querySelectorAll("div")
+for (i  = 0; i < nodes.length; ++i) {
+    var position = nodes[i].style.getPropertyValue("position")
+    if (position != null && position.indexOf("relative") != -1) {
+        nodes[i].style.setProperty("display", "none")
+    }
+}
+</script>
 </body></html>
 <?php
 	wfRestoreWarnings();
-- 
1.6.1

